docs/best_practices.md

```diff
@@ -11,46 +11,56 @@ Links should point to specific commits, and not a branch (in case the branch or
-- [Write unit tests for new components](#write-unit-tests-for-new-components)
-  - [Manipulating time in unit tests](#manipulating-time-in-unit-tests)
-- [Add smoke tests for new routes](#add-smoke-tests-for-new-routes)
-- [Adding global script tags](#adding-global-script-tags)
+- [Current Tech Stack](#current-tech-stack)
+- [TypeScript](#typescript)
+- [UI](#ui)
+- [Routing](#routing)
+- [Relay](#relay)
+- [Code organization](#Code-organization)
+- [Code style](#code-style)
+- [Testing](#testing)
 - [Tracking Events](#tracking-events)
 
-## Current Preferred Practices
+## Current Tech Stack
 
 The app is currently written responsively in React. Server-side code is built on top of Express.js; however, most server-side needs have been abstracted away by our framework (see below).
 
+## TypeScript
+
+We use [TypeScript](https://www.typescriptlang.org/docs) to maximize runtime code safety and prevent runtime bugs.
+
+### Avoid copying and try to fix `// @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION` flags
+
+Around mid-2021 we migrated to strict type checking for **all new code**. What this meant in practice was that all _old code_ that failed strict type checking was silenced via a special flag inserted by a script (`// @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION`) with all _new code_ expected to adhere to best practices. Going forward, this flag should never be used, and if encounted while working on old code it should be removed and the type error fixed.
+
+## UI
+
 ### When writing UI, use Palette
 
-Our design system [Palette](https://github.com/artsy/palette) is used for most UI needs. Check out [the docs](https://palette.artsy.net/)!
+Our design system [Palette](https://github.com/artsy/palette) is used for most UI needs.
+
+- Check out [the docs](https://palette.artsy.net/).
+- Checkout the Storybook including components in accordance with the most recent design system [here](https://palette-storybook.artsy.net/).
 
 > If interested in some of the lower-level particulars of how Palette was built, see the [styled-system](https://styled-system.com/) docs.
 
-### For routing, use our framework
+### Commonly used components
 
-Individual sub-apps (represented by routes like `/artist/:id` or `/collect`) are built on top of a home-grown SSR (server-side-rendering) framework based around [Found](https://github.com/4Catalyzer/found), a routing library. Sub-apps are mounted within the [global routes file](https://github.com/artsy/force/blob/main/src/routes.tsx).
+- [Box](https://palette.artsy.net/elements/box/) - is a div which we can pass extra props to, like margins for example. The content is displayed vertically.
 
-To learn how to create a new sub-app, see [the docs](https://github.com/artsy/force/blob/1842553ad34475bc3b804f00c6410d7f23d64f65/docs/adding_new_app.md).
+- [Flex](https://palette.artsy.net/elements/flex/) - It is a Box with display=flex . The content is displayed vertically horizontally.
```

Comment:
"The content is displayed vertically horizontally" — typo? But also not a useful sentence to add. The default `flex-direction` of a `div` that is set to `display: flex` is `row`

Reply:
https://github.com/artsy/force/pull/11512/files